src/Context/ApiContext.php

```diff
@@ -273,15 +273,19 @@ private function dropSessionContext()
      */
     public function ensureSessionActive()
     {
-        if (is_null($this->sessionContext)) {
-            return;
+        if (!is_null($this->sessionContext) && $this->isSessionExpired()) {
```

Comment:
check for `is_null($this->sessionContext)` can go from here, because you do it in `isSessionExpired` already :)

Reply:
If you follow my suggestion and change `isSessionExpired` to `isSessionActive` or `hasActiveSession`, Please do not forget to add an exclamation mark here (only reset if NOT has an active session :))

Reply:
Mhmm AFAIK if we do this then if session is null it will run in a infinity loop resetting the session. At least that was my expiernce in Java 🤔 will double check to make sure.

Reply:
Ah yea! in JAVA we will have infinite loop, and in PHP
```
 $timeExpiry = $this->sessionContext->getExpiryTime()->getTimestamp();
```

This will cause an error as of `sessionContext` is null! So therefore we need `!is_null($this->sessionContext)` in `ensureSessionActive` still 🤔. The null `sessionContext` will get handled higher up the code before a request is made.

Reply:
Ah but then based on your suggestion below this can be fixed 👍